src/models/Voucher.js

```diff
@@ -0,0 +1,57 @@
+import mongoose from 'mongoose';
+
+// schema Voucher variables
+const DOCUMENT_NAME = 'Voucher';
+const COLLECTION_NAME = 'Vouchers';
+
+const voucherSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    code: {
+      type: String,
+      required: true,
+      unique: true,
+    },
+    max_usage: {
+      type: Number,
+      required: true,
+    },
+    duration: {
+      type: Number,
+      required: true,
+    },
+    discount: {
+      type: Number,
+      required: true,
+    },
+    discount_types: {
+      type: String,
+      enum: ['percentage', 'fixed'],
+    },
+    minimum_order_price: {
+      type: Number,
+    },
+    voucher_type: [
```

Comment:
chỗ này là quan hệ 1-1. 1 voucher chỉ có 1 voucher_type. Bạn sửa lại và tính toán lại xử lí logic trong service